Framework/DataHandling/src/LoadCanSAS1D.cpp

```diff
@@ -34,16 +35,15 @@ using namespace Mantid::API;
 using namespace Mantid::DataObjects;
 
 namespace {
-int getGeometryID(const std::string &selection) {
+int getGeometryID(std::string selection) {
```

Comment:
We can't guarantee `constness` in this method anymore as `tolower` modifies the string in place. As a result, passing the value by reference would introduce hidden behaviour as it would get you the geometry ID, but it would also modify the string you passed in to make it lowercase unless we made a copy inside anyway. 

Neither of these are ideal, so setting the `selection` string to be passed by value means that you are forced to use `std::move` at the call point. You therefore give up your right to use it in the caller, and we can modify it in `getGeometryID()` without worrying about it causing issues elsewhere.

Reply:
Ah, thanks, that makes sense - I hadn't thought properly about the effect of the conversion to lowercase.